analyzer/operations.go

```diff
@@ -302,6 +345,161 @@ func MatchChangesOnSubAccount(op1 *Operation, op2 *Operation, subAccountType Sub
 	return reflect.DeepEqual(op1Changes, op2Changes)
 }
 
+func GetTransferChangesWithTobinTax(from, to common.Address, value *big.Int, tobinTax *TobinTax) []BalanceChange {
+	taxAmount, afterTaxAmount := tobinTax.Apply(value)
+
+	changes := GetTransferChanges(from, to, afterTaxAmount)
+
+	if taxAmount.Cmp(utils.Big0) > 0 {
+		changes = append(changes, GetTransferChanges(from, tobinTax.Recipient, taxAmount)...)
+	}
+
+	return changes
+}
+
+func GetTransferChanges(from, to common.Address, value *big.Int) []BalanceChange {
+	return []BalanceChange{
+		{Account: NewAccount(from, AccMain), Amount: negate(value)},
+		{Account: NewAccount(to, AccMain), Amount: value},
+	}
+}
+
+// ReconcileLockedGoldTransfers merges transfer and lockedGold operations
+// together into one slice that contains no duplicate operations.
+//
+// See NewTransfer, NewWithdrawal, NewSlash for examples.
+func ReconcileLockedGoldTransfers(lockedGoldOps, transferOps []Operation, tobinTax *TobinTax, lockedGold common.Address) ([]Operation, error) {
+	ops := make([]Operation, 0, len(transferOps)+len(lockedGoldOps))
+
+	reconcile := func(lgOp *Operation, trOp Operation) {
+		// start with the transferOp
+		op := trOp
+		// change type to OpLockGold because we will be adding locked gold specific balance changes
+		op.Type = OpLockGold
+		// add all changes from lgOp with SubAccountTypes specficic to LockedGold
+		for _, bc := range lgOp.Changes {
+			if bc.Account.SubAccount.Identifier != AccMain {
+				op.Changes = append(op.Changes, bc)
+			}
+		}
+		// replace lgOp with reconciled op
+		*lgOp = op
+	}
+
+	findMatchAndReconcile := func(transferOps []Operation, lgOp *Operation, i int) (int, error) {
+		for ; i < len(transferOps); i++ {
+			if MatchChangesOnSubAccount(lgOp, &transferOps[i], AccMain) {
```

Comment:
you probably don't need both.... since either you have a >0 tobin tax or you don't... maybe have a single function to do the matching